### PR TITLE
ENH: Check matching inputs/outputs in umath generation

### PR DESCRIPTION
**Pull request by aragilar**
This is a small patch which flags typos and similar problems when modifying the umath subsystem.

### CODE REVIEW

File: numpy/core/code_generators/generate_umath.py [hunk 1]
```diff
@@ -97,6 +97,8 @@ def TD(types, f=None, astype=None, in_=None, out=None, simd=None):
         out = (out,) * len(types)
     elif out is None:
         out = (None,) * len(types)
+    if len(in_) != len(out):
+        raise RuntimeError("Number of inputs don't match outputs")
```
Comment (eric-wieser) [inline numpy/core/code_generators/generate_umath.py]: Would be better as
```
elif len(out) != len(types):
    raise ValueError(...)
```
and same for the inputs

Comment (aragilar) [reply numpy/core/code_generators/generate_umath.py]: Yeah, that makes far more sense.

**Review by eric-wieser — APPROVED** [review]
LGTM, waiting on tests to pass

**Comment by aragilar** [comment]
I've updated to checking against types, and replaced an assertion above with an explicit test (which is safer in case the assertion gets optimised out).

**Comment by eric-wieser** [comment]
Thanks @aragilar!

**Pull request merged**